extensions/wikia/SpecialDiscussions/SpecialDiscussionsController.class.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+use Swagger\Client\ApiException;
+use Swagger\Client\Discussion\Api\SitesApi;
+use Wikia\DependencyInjection\Injector;
+use Wikia\Service\Swagger\ApiProvider;
+
+/**
+ * Discussion user log page
+ */
+class SpecialDiscussionsController extends WikiaSpecialPageController {
+
+	const HTTP_STATUS_OK = 200;
+	const DISCUSSIONS_ACTION = 'specialdiscussions';
+	const SERVICE_NAME = 'discussion';
+	const TIMEOUT = 5;
+	const EDIT_TOKEN = 'editToken';
+
+	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+
+	private $logger;
+	private $sitesApi;
+	private $siteId;
+	private $siteName;
+
+	public function __construct() {
+		parent::__construct( 'Discussions', '', false );
+		$this->sitesApi = $this->getDiscussionSitesApi();
+		$this->siteId = F::app()->wg->CityId;
+		$this->siteName = WikiFactory::getVarValueByName( 'wgSitename', $this->siteId );
+		$this->logger = Wikia\Logger\WikiaLogger::instance();
+	}
+
+	public function index() {
+		$this->checkAccess();
+
+		$this->setHeaders();
+		$this->response->addAsset( 'special_discussions_scss' );
+
+		$this->wg->Out->setPageTitle( wfMessage( 'discussions-pagetitle' )->escaped() );
+
+		$editToken = $this->getVal( self::EDIT_TOKEN );
+		if ( !empty( $editToken ) ) {
+			$this->activateDiscussions();
+		}
+
+		$this->setIndexOutput();
+	}
+
+	public function inputForm() {
+		$this->checkAccess();
```

Comment:
The `checkAccess` method isn't meant to keep people from doing something they shouldn't be doing (that's what `assertValidPostRequest` is for) but to remind non-staff that there is nothing here for them.  For the that reason I don't think it's necessary to call it here and in the `discussionsLink` method.  Its already been called in `index` and the only way around that would be to URL hack to hit `inputForm` directly.

Alternately, the call to `checkAccess` could be removed from all these controller methods (including `index`) and added to `__construct`.  While it would still get called twice in this case because of the selfRequest stuff, it would mean it would never need to be added (or forgotten) to any new controller methods that are expected to be called by the client.